### DISCUSSION TITLE
how to register users

### DISCUSSION THREAD

**Discussion by gy95** [dexidp/dex#2024] [Q&A]
does dex support self-registration? i only find third party register entrance,such as github, google, etc., but don't find way to register a user myself.

**Comment by gy95** [comment]
how to enable registration?
i don't find registration button, i only find button to login with github or google.

**Reply by nabokihms** [reply]
Hello! If you do not want to use third-party providers, dex has [GRPC API](https://dexidp.io/docs/api/) to register and verify users.
These users have some limitations, e.g., no groups, no preferred username.
